### PR TITLE
[make/en] Adding a tutorial on (gnu) make

### PR DESCRIPTION
**Pull request by robochat**
Here is a first draft of a make tutorial. I'm not an expert at make at all so corrections are probably needed!

### CODE REVIEW

**Comment by vendethiel** [comment]
Looks like a good idea... And there's a lot of ground to cover, as well!

File: make.html.markdown [hunk 1]
```diff
@@ -0,0 +1,236 @@
+---
+language: make
+contributors:
+    - ["Robert Steed", "https://github.com/robochat"]
+filename: Makefile
+---
+
+A Makefile defines a graph of rules for creating a target (or targets).
+It's purpose is the do the minimum amount of work needed to update a
```
Comment (vendethiel) [inline make.html.markdown]: Its*
to do*

Comment (robochat) [inline make.html.markdown]: I found another it's error too. Embarrassing.

Comment (switchhax) [inline make.html.markdown]: Well it´s always going to be reviewed, otherwise you can fix errors also later :)

**Comment by geoffliu** [comment]
I forget, does the GNU variant support, for example, `.c.o` rules?

**Comment by robochat** [comment]
I think that it's supported but I think that they suggest that pattern rules should be used in preference.

**Comment by geoffliu** [comment]
I think it's still worth a mention. I've definitely seen those rules in the wild before, so I can imagine someone needing to know what those are. Nothing extensive, even if you just mention the name and what the look like, so someone has a word to Google search if they need to.

**Comment by geoffliu** [comment]
Thanks, this looks great!

**Comment by robochat** [comment]
Great!

**Pull request merged**